netqueue.go

```diff
@@ -101,91 +125,107 @@ func (q *NetfilterQueue) iptables(flag string) {
 	}
 }
 
-func (q *NetfilterQueue) loop() {
-	if len(q.IPs) == 0 {
-		return
-	}
-	queue, err := nfqueue.NewNFQueue(uint16(q.Number), maxPacketsInQueue, nfqueue.NF_DEFAULT_PACKET_SIZE)
-	if err != nil {
-		panic(err)
-	}
+func (q *netfilterQueue) loop(queue *nfqueue.NFQueue, ctx context.Context) {
 	defer queue.Close()
+	defer close(q.capture)
+	defer close(q.capturing)
+	defer close(q.release)
 
 	procNf, err := ReadProcNetfilter()
 	if err != nil {
 		panic(err)
 	}
 
-	accepting := true
-	accept := sync.NewCond(&sync.Mutex{})
-	accept.L.Lock()
+	lastQueueDropped := uint(0)
+	lastUserDropped := uint(0)
```

Comment:
This method is private of an only queue object, so each queue will have its own loop with its own set of variables.